app/components/wds-global-navigation/search.js

```diff
@@ -29,7 +59,247 @@ export default Component.extend({
+
+		/**
+		 * This was queued to run before the user has finished typing, and when they
+		 * finished typing it may have turned out that they were just backspacing OR
+		 * they finished typing something that was already in the cache, in which case
+		 * we just ignore this request because the search fn already put the cached
+		 * value into the window.
+		 */
+		if (!query || this.hasCachedResult(query) || this.requestInProgress(query) || this.get('isDestroyed')) {
+			return;
+		}
+
+		this.startedRequest(query);
+
+		fetch(uri)
+			.then((response) => {
+				if (response.ok) {
+					if (this.get('isDestroyed')) {
+						return;
+					}
+
+					return response.json().then((data) => {
+						const suggestions = data.suggestions.map((suggestion) => {
+							return EmberObject.create({
+								title: suggestion
+							});
+						});
+
+						/**
+						 * If the user makes one request, request A, and then keeps typing to make
+						 * request B, but request A takes a long time while request B returns quickly,
+						 * then we don't want request A to dump its info into the window after B has
+						 * already inserted the relevant information.
+						 * Also, we don't want to show the suggestion results after a real search
+						 * will be finished, what will happen if search request is still in progress.
+						 */
+						if (!this.get('searchRequestInProgress') && query === this.get('query')) {
+							this.setSearchSuggestionItems(suggestions);
+						}
+
+						this.cacheResult(query, suggestions);
+					});
+				} else if (response.status === 404) {
+					// When we get a 404, it means there were no results
+					if (query === this.get('query')) {
+						this.setSearchSuggestionItems();
+					}
+
+					this.cacheResult(query);
+				} else {
+					this.get('logger').error('Search suggestions error', response);
+				}
+			})
+			.catch((reason) => this.get('logger').error('Search suggestions error', reason))
+			.finally(() => {
+				// We have a response, so we're no longer loading the results
+				if (query === this.get('query') && !this.get('isDestroyed')) {
+					this.set('isLoadingResultsSuggestions', false);
+				}
+
+				this.endedRequest(query);
+			});
+	},
+
+	/**
+	 * Wrapper for search suggestions performing, that also checks the cache
+	 */
+	getSuggestions(query) {
+		if (this.get('isDestroyed')) {
+			return;
+		}
+
+		this.setProperties({
+			suggestions: [],
+			searchRequestInProgress: false
+		});
+
+		// If the query string is empty or shorter than the minimal length, return to leave the view blank
+		if (!query || query.length < this.get('queryMinimalLength')) {
+			this.set('isLoadingResultsSuggestions', false);
+		} else if (this.hasCachedResult(query)) {
+			this.setSearchSuggestionItems(this.getCachedResult(query));
+		} else {
+			this.set('isLoadingResultsSuggestions', true);
+			run.debounce(this, () => this.requestSuggestionsFromAPI(query), this.get('debounceDuration'));
```

Comment:
Does it work? I guess it will create new instance of debounce function each time `getSuggestions` is called. I think we should extrac `() => this.requestSuggestionsFromAPI(query)` to component method and pass this method as an argument of debounce here. But I may be wrong ;)

Reply:
Fixed!